thorlcr/activities/csvread/thcsvrslave.cpp

```diff
@@ -146,9 +148,11 @@ class CCsvReadSlaveActivity : public CDiskReadSlaveActivityBase
         }
         virtual void close(CRC32 &fileCRC)
         {
-            CriticalBlock block(statsCs);
-            mergeStats(fileStats, iFileIO);
-            iFileIO.clear();
+            {
+                CriticalBlock block(statsCs);
+                mergeStats(fileStats, iFileIO);
+                iFileIO.clear();
```

Comment:
Ideally this should assign iFileIO to a variable declared outside the block, so that the close isn't within the critical section.